sol-did/programs/sol-did/src/state/did_account.rs

```diff
@@ -194,15 +257,21 @@ impl DidAccount {
 
     /// Returns true if `other` is a valid controller of this DID
     pub fn is_directly_controlled_by(&self, other: &DidAccount) -> bool {
-        other.other_controllers.contains(&self.to_string())
+        let this_did = normalize_did_cluster(self.to_string().as_str());
+        other
+            .other_controllers
```

Comment:
I'm trying to understand. Why do we even need to support `other_controller`? All `did:sol` controllers are in `native_controllers`

Reply:
fixed